resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -745,9 +745,11 @@ private[spark] class Client(
     val nodeUrls = kubernetesClient.nodes.list.getItems.asScala
       .filterNot(node => node.getSpec.getUnschedulable != null &&
         node.getSpec.getUnschedulable)
-      .flatMap(_.getStatus.getAddresses.asScala.map(address => {
+      .flatMap(_.getStatus.getAddresses.asScala)
+      .filter(_.getType == "ExternalIP")
```

Comment:
A quick comment here would be nice. Out of curiosity, is there a reference for this constant somewhere?